rust/src/io/local.rs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fs::File;
+use std::ops::Range;
+use std::sync::Arc;
+// TODO: worry about windows later.
+use std::os::unix::fs::FileExt;
+
+use async_trait::async_trait;
+use bytes::{Bytes, BytesMut};
+use object_store::path::Path;
+
+use super::object_reader::ObjectReader;
+use crate::Result;
+
+pub struct LocalObjectReader {
+    file: Arc<File>,
+}
+
+impl LocalObjectReader {
+    pub fn open(path: &Path) -> Result<Box<dyn ObjectReader>> {
+        let local_path = format!("/{}", path.to_string());
```

Comment:
so relative paths won't work?

Reply:
The relative path is handled via the `ObjectStore` to keep it consistent with the current `ObjectStore` API, which takes `bucket` as part of `ObjectStore` instead of `Object Path`.

https://github.com/eto-ai/lance/blob/main/rust/src/io/object_store.rs#L65